airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -321,6 +320,14 @@ def set_autocommit(self, conn, autocommit: Any) -> None:
     def get_autocommit(self, conn):
         return getattr(conn, "autocommit_mode", False)
 
+    @staticmethod
+    def split_sql_string(sql: str) -> list[str]:
+        split_statements_tuple = util_text.split_statements(StringIO(sql))
+        return [sql_string for sql_string, _ in split_statements_tuple if sql_string]
+
+    def _update_query_ids(self, cursor) -> None:
+        self.query_ids.append(cursor.sfqid)
+
     def run(
```

Comment:
This method still implemented here to keep Snowflake's `split_statements: bool = True` default parameter